File: Python/소프티어/스마트 물류.py
import sys

n, k = list(map(int,sys.stdin.readline().split()))
data = list(sys.stdin.readline().rstrip())

answer = 0
for i in range(0,len(data)):
    if data[i] == 'P':
        for j in range(i-k,i+k+1):
            if j<0 or j>=n:
                continue
            elif data[j] == 'H':
                answer += 1
                data[j] = '-'
                break

print(answer)

File: AlgorithmReview/softeer/스마트 물류.swift
//
//  스마트 물류.swift
//  AlgorithmReview
//
//  Created by 박소윤 on 2024/02/02.
//

import Foundation

let input = readLine()!.split(separator: " ").map{ Int($0)! }
let (n, k) = (input[0], input[1])

var data = Array(readLine()!)

var answer = 0
for i in 0..<n {
    if data[i] != "P" {
        continue
    }
    for j in i-k...i+k {
        if j < 0 || j >= n {
            continue
        }
        else if data[j] == "H"{
            answer += 1
            data[j] = "-"
            break
        }
    }
}

print(answer)

/*
20 1
HHPHPPHHPPHPPPHPHPHP
 */

/*
20 2
HHHHHPPPPPHPHPHPHHHP
*/

